server/etcdmain/config.go

```diff
@@ -76,8 +78,12 @@ type configFlags struct {
 }
 
 func newConfig() *config {
+	return newConfigFromEmbedConfig(embed.NewConfig())
+}
+
+func newConfigFromEmbedConfig(ec *embed.Config) *config {
```

Comment:
The implementation of `newConfig` is simple enough, let's try not to do such unnecessary change.

Reply:
done.